src/pages/views.py

```diff
@@ -265,21 +252,63 @@ def get_context_data(self, **kwargs):
         periods = list(self.clusive_user.periods.all())
         current_period = self.clusive_user.current_period
         assigned = self.view == 'assigned'
+        is_teacher = self.clusive_user.can_manage_periods
+        is_guest = self.clusive_user.role == 'GU'
+
+        if is_teacher:
+            readings = get_popular_reads_data(self.clusive_user, periods,
+                                              current_period, assigned_only=assigned)
+        else:
+            # Student
+            if not is_guest:
+                if self.view == 'assigned' or self.view == 'popular':
+                    readings = get_popular_reads_data(self.clusive_user, periods,
+                                                      current_period, assigned_only=assigned)
+                else:
+                    # self.view == 'recent'
+                    readings = get_recent_reads_data(self.clusive_user)
+            # Guest
+            else:
+                if self.view == 'assigned':
+                    # "Clues to Clusive"
+                    readings = {
+                        'view': 'assigned',
+                        'all': Book.get_featured_books()[:3]
+                    }
+                elif self.view == 'popular':
+                    readings = get_popular_reads_data(self.clusive_user, periods,
+                                                      current_period, assigned_only=False)
+                else:
+                    # self.view == 'recent'
+                    readings = get_recent_reads_data(self.clusive_user)
+
         context.update({
+            'is_teacher': is_teacher,
+            'is_guest': is_guest,
             'current_period': current_period,
-            'period_name': current_period.name,
+            'period_name': current_period.name if current_period else None,
             'query': None,
-            'data': get_popular_reads_data(self.clusive_user, periods, current_period, assigned_only=assigned),
+            'data': readings,
         })
         return context
 
 
+def get_recent_reads_data(clusive_user):
+    return {
+        'view': 'recent',
+        'total': Paradata.latest_for_user(clusive_user),
```

Comment:
I think 'all' should be set here, and 'total' set to a number. As is, 'recent' view is always empty.